cryodrgn/commands/preprocess.py

```diff
@@ -106,12 +117,30 @@ def preprocess(imgs):
             ret = fft.symmetrize_ht(ret)
         return ret
 
+    def preprocess_cupy(imgs):
+        if lazy:
+            imgs = _combine_imgs(imgs)
+            imgs = cp.concatenate([cp.asarray(i.get()) for i in imgs])
+        if window:
+            imgs *= dataset.window_mask(original_D, args.window_r, .99, use_cupy=True)
+            
+        ret = cp.asarray([fft.ht2_center(img) for img in imgs])
+        if invert_data:
+            ret *= -1 
+        if downsample:
+            ret = ret[:, start:stop, start:stop]
+        ret = fft.symmetrize_ht(ret)
+        return ret    
+
     def preprocess_in_batches(imgs, b):
         ret = np.empty((len(imgs), D+1, D+1), dtype=np.float32)
         Nbatches = math.ceil(len(imgs)/b)
         for ii in range(Nbatches):
             log(f'Processing batch of {b} images ({ii+1} of {Nbatches})')
-            ret[ii*b:(ii+1)*b,:,:] = preprocess(imgs[ii*b:(ii+1)*b])
+            if args.use_cupy:
```

Comment:
Instead of reaching out and looking at the global `args` inside this function, let's pass a `use_cupy` argument, and use `args.use_cupy` in the caller. I see that we're following this pattern in some other places. I'll make a note to fix those occurrences too.